compiler/rustc_privacy/src/lib.rs

```diff
@@ -422,13 +439,15 @@ impl VisibilityLike for Option<EffectiveVisibility> {
     // (which require reaching the `DefId`s in them).
     const SHALLOW: bool = true;
     fn new_min(find: &FindMin<'_, '_, Self>, def_id: LocalDefId) -> Self {
-        if let Some(min) = find.min {
-            return find
-                .effective_visibilities
-                .effective_vis(def_id)
-                .map(|eff_vis| min.min(*eff_vis, find.tcx));
-        }
-        None
+        let private_vis = ty::Visibility::Restricted(find.tcx.parent_module_from_def_id(def_id));
+
+        let effective_vis = find
+            .effective_visibilities
+            .effective_vis(def_id)
+            .cloned()
+            .unwrap_or(EffectiveVisibility::from_vis(private_vis));
```

Comment:
```suggestion
            .unwrap_or_else(|| EffectiveVisibility::from_vis(ty::Visibility::Restricted(find.tcx.parent_module_from_def_id(def_id))));
```
No need to calculate the private visibility eagerly.